tests/func/artifacts/test_artifacts.py

```diff
@@ -206,3 +207,55 @@ def test_parametrized(tmp_dir, dvc):
     assert tmp_dir.dvc.artifacts.read() == {
         "dvc.yaml": {"myart": Artifact(path="myart.pkl", type="model")}
     }
+
+
+def test_get_path_subrepo(tmp_dir, scm, dvc):
+    subrepo = tmp_dir / "subrepo"
+    make_subrepo(subrepo, scm)
+    (subrepo / "dvc.yaml").dump(dvcyaml)
+
+    assert dvc.artifacts.get_path("subrepo:myart") == os.path.join(
+        "subrepo", "myart.pkl"
+    )
+    assert dvc.artifacts.get_path("subrepo/dvc.yaml:myart") == os.path.join(
+        "subrepo", "myart.pkl"
+    )
+
+    assert subrepo.dvc.artifacts.get_path("subrepo:myart") == os.path.join(
+        "subrepo", "myart.pkl"
+    )
+    assert subrepo.dvc.artifacts.get_path("subrepo/dvc.yaml:myart") == os.path.join(
+        "subrepo", "myart.pkl"
+    )
+
+
+@pytest.mark.parametrize(
+    "repo",
+    [
+        None,
+        "{tmp_dir}",
+        "file://{tmp_posixpath}",
+        "{subrepo}",
+    ],
+)
+def test_artifacts_download_subrepo(tmp_dir, scm, repo):
+    subrepo = tmp_dir / "subrepo"
+    make_subrepo(subrepo, scm)
+    dvc = subrepo.dvc
+
+    artifact = Artifact(path="myart.txt", type="model")
+    subrepo.scm_gen("myart.txt", "hello world", commit="add myart.txt")
+    dvc.artifacts.add("myart", artifact, dvcfile=os.fspath(subrepo / "dvc.yaml"))
+    scm.add_commit([subrepo / "dvc.yaml"], message="add dvc.yaml")
+    scm.tag("subrepo=myart@v2.0.0", annotated=True, message="foo")
+
+    if repo:
+        repo = repo.format(
+            tmp_dir=tmp_dir,
+            tmp_posixpath=tmp_dir.as_posix(),
+            subrepo=subrepo,
+        )
+
+    assert not (tmp_dir / "myart.txt").exists()
+    assert dvc.artifacts.get(repo, "subrepo:myart") == (1, "myart.txt")
```

Comment:
What about `dvc.artifacts.get(subrepo, "myart")`?

Reply:
With the current change, the name (and the `tag` format) is always required to be relative to the git root. So they don't change with respective to the git repo.



```suggestion
    assert dvc.artifacts.get(subrepo, "subrepo:myart") == (1, "myart.txt")
```

Reply:
Thanks. So if I start from the parent repo, these are the same:

```
$ dvc artifacts get . subrepo:myart
$ dvc artifacts get subrepo subrepo:myart
```

But I can't get the artifact if I start from within the subrepo. Both of these fail:

```
$ dvc artifacts get . subrepo:myart
$ dvc artifacts get . myart
```

Reply:
It might be fine to even fail if we can raise a useful error to explain how to fix it.